src/PoolManager.sol

```diff
@@ -294,6 +294,14 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         _accountDelta(currency, paid.toInt128(), msg.sender);
     }
 
+    /// @inheritdoc IPoolManager
+    function clear(Currency currency, uint256 amount) external onlyWhenUnlocked {
+        int256 current = currency.getDelta(msg.sender);
+        if (current < 0) revert OwedFundsCannotBeCleared();
+        if (amount != uint256(current)) revert MustClearExactBalance();
```

Comment:
These 2 lines feel inefficient. It checks that current isnt negative, and then it requires it be equal to `amount`.

If we know that `amount` is positive (it has to be as its a uint256), then we know that if `amount==current` then current cannot be <0 either.

I think it would be more efficient to just do

```
        int256 current = currency.getDelta(msg.sender);
        int128 amountDelta = amount.toInt128();
        if (amountDelta != current) revert MustClearExactBalance();
        _accountDelta(currency, -amountDelta, msg.sender);
```